### PR TITLE
Debug for recipe collisions

### PR DESCRIPTION
**Pull request by chochem**
Adds info to the GT log for recipe collisions when in recipe debug mode. (java arg: -Dgt.recipebuilder.debug=true)

### CODE REVIEW

File: src/main/java/gregtech/api/util/GT_RecipeBuilder.java [hunk 1]
```diff
@@ -126,6 +134,17 @@ private static void handleNullRecipeComponents(String componentType) {
         }
     }
 
+    public static void handleRecipeCollision(String details) {
+        if (PANIC_MODE) {
+            throw new IllegalArgumentException("Recipe Collision");
```
Comment (miozune) [inline src/main/java/gregtech/api/util/GT_RecipeBuilder.java:139]: Also logging `details` here would be better than just killing program

Comment (chochem) [reply src/main/java/gregtech/api/util/GT_RecipeBuilder.java:139]: I dont know, boubs says it should crash in panic mode

Comment (boubou19) [reply src/main/java/gregtech/api/util/GT_RecipeBuilder.java:139]: for me, panic mode should be enabled on production only, so we let the user knows their custom recipes are badly formed. Then we must kill all the existing conflicts before releasing stable.

Comment (miozune) [reply src/main/java/gregtech/api/util/GT_RecipeBuilder.java:139]: I mean log _and_ crash. Stacktrace is sometimes not useful to find which recipe had collision

Comment (chochem) [reply src/main/java/gregtech/api/util/GT_RecipeBuilder.java:141]: like this?

Comment (miozune) [reply src/main/java/gregtech/api/util/GT_RecipeBuilder.java:141]: Yes. Also early return (`if (!PANIC_MODE && !DEBUG_MODE) return;`) can remove duplicated code

Comment (chochem) [reply src/main/java/gregtech/api/util/GT_RecipeBuilder.java:144]: done

**Pull request merged**